mavros/src/plugins/setpoint_position.cpp

```diff
@@ -181,9 +185,43 @@ class SetpointPositionPlugin : public plugin::PluginBase,
 	}
 
 	/**
-	 * Gets gps setpoint, converts it to local ENU, and sends it to FCU
+	 * Gets setpoint position setpoint and send SET_POSITION_TARGET_GLOBAL_INT
 	 */
 	void setpointg_cb(const mavros_msgs::GlobalPositionTarget::ConstPtr &req)
+	{
+		Eigen::Vector3d velocity, af;
+		float yaw, yaw_rate;
+
+		tf::vectorMsgToEigen(req->velocity, velocity);
+		tf::vectorMsgToEigen(req->acceleration_or_force, af);
+
+		// Transform frame ENU->NED
+		velocity = ftf::transform_frame_enu_ned(velocity);
+		af = ftf::transform_frame_enu_ned(af);
+		yaw = ftf::quaternion_get_yaw(
+					ftf::transform_orientation_aircraft_baselink(
+						ftf::transform_orientation_ned_enu(
+							ftf::quaternion_from_rpy(0.0, 0.0, req->yaw))));
+		Eigen::Vector3d ang_vel_enu(0.0, 0.0, req->yaw_rate);
+		auto ang_vel_ned = ftf::transform_frame_ned_enu(ang_vel_enu);
+		yaw_rate = ang_vel_ned.z();
+
+		set_position_target_global_int(
+					req->header.stamp.toNSec() / 1000000,
+					req->coordinate_frame,
+					req->type_mask,
+					req->latitude * 1e7,
+					req->longitude * 1e7,
+					req->altitude,
+					velocity,
+					af,
+					yaw, yaw_rate);
+	}
+
+	/**
+	 * Gets gps setpoint, converts it to local ENU, and sends it to FCU
+	 */
+	void setpointg2l_cb(const mavros_msgs::GlobalPositionTarget::ConstPtr &req)
```

Comment:
@Jaeyoung-Lim now that I see this, this callback msg type doesn't make sense as well. Can you change it to `GeoPoseStamped` as well? Thanks!

Reply:
Done!